Tools/cases_generator/analyzer.py

```diff
@@ -55,10 +57,16 @@ def from_list(properties: list["Properties"]) -> "Properties":
             passthrough=all(p.passthrough for p in properties),
         )
 
+    @property
+    def infallible(self) -> bool:
+        return not self.pop_error and not self.no_pop_error
```

Comment:
Again, the name `no_pop_error` makes this condition really annoying to mentally parse (it *reads* like it should be impossible).

Reply:
Renamed to `error_with_pop` and `error_without_pop`.